topics/metabolomics/tutorials/gc_ms_with_xcms/tutorial.md

```diff
@@ -0,0 +1,482 @@
+>    - {% icon param-collection %} *"Input unrefined MS data"*: `input` (Input dataset collection)
+>    - *"Do you agree to the vendor licenses?"*: `Yes`
+>    - *"Output Type"*: `mzML`
+>    - In *"Data Processing Filters"*:
+>        - *"Apply peak picking?"*: `Yes` (This option will compute centroids in the m/z domain.)
+>
+>    > <comment-title> Centroids </comment-title>
+>    >
+>    > `msconvert` with selected parameters computes centroids in the m/z domain. MS instruments continuously sample and record signals. Therefore, a mass peak for a single ion in one spectrum consists of multiple intensities at discrete m/z values. Centroiding is the process of reducing these mass peaks to a single representative signal, the centroid. This results in much smaller file sizes without losing too much information. In the further steps, **XCMS** uses the _centWave_ chromatographic peak detection algorithm, which was designed for centroided data. That is the reason we perform the centroiding prior to chromatographic peak detection in this step.
+>    {: .comment}
+>
+{: .hands_on}
+
+## Create the XCMS object
+
+The first part of data processing is using the **XCMS** tool to detect peaks in the MS signal. For that, we first need to take the `.mzML` files and create a format usable by the **XCMS** tool. {% tool [MSnbase readMSData](toolshed.g2.bx.psu.edu/repos/lecorguille/msnbase_readmsdata/msnbase_readmsdata/2.16.1+galaxy0) %} ({% cite gatto2012msnbase %}. {% cite gatto2020msnbase %}) takes as input our files and prepares `RData` files for the first **XCMS** step.
+
+> <hands-on-title> Create the **XCMS** object </hands-on-title>
+>
+> 1. {% tool [MSnbase readMSData](toolshed.g2.bx.psu.edu/repos/lecorguille/msnbase_readmsdata/msnbase_readmsdata/2.16.1+galaxy0) %} with the following parameters:
+>    - {% icon param-file %} *"File(s) from your history containing your chromatograms"*: `input.mzML` (output of **msconvert** {% icon tool %})
+>
+>    {% snippet faqs/galaxy/tools_select_collection.md %}
+>
+>    > <comment-title> Output - `input.raw.RData` </comment-title>
+>    >
+>    > Collection of `rdata.msnbase.raw` files. `Rdata` file that is necessary in the next step of the workflow. These serve for an internal R representation of **XCMS** objects needed in the further steps.
+>    {: .comment}
+{: .hands_on}
+
+# Peak detection using XCMS
+
+The first step in the workflow is to detect the peaks in our data using **XCMS**. This part, however, is covered by a [separate tutorial]({{ site.baseurl }}/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html). Although the tutorial is dedicated to LC-MS data, it can also be followed for our GC-MS data. Therefore, in this section, we do not explain this part of the workflow in detail but rather refer the reader to the dedicated tutorial. Please also pay attention to the parameter values for individual Galaxy tools, as these can differ from the referred tutorial and are adjusted to our dataset.
```

Comment:
I'm not sure what you mean by "without waiting for the results". Do you mean we could provide the data already processed by XCMS and allow the trainee to upload them and effectively skip XCMS part? Or did you have something else in mind?

@hechth what is your opinion?

Reply:
About this point, the idea that I did not expressed well was that since the XCMS processing part can be a bit long to run, and since you refered to the LC-MS XCMS tutorial, it may have been interesting to warn the trainee about the time needed to run the following steps and thus to invite him/her not to wait to launch the XCMS part of the GC-MS tutorial (that can be launch all in a raw) even if he/she plans to take a look ad the LC-MS tutorial (the idea was more to optimise the learning time with a "reading while it runs" approach). Sorry for the potential misunderstanding.

Anyway, it is a good idea to provide a final XCMS object to save time so thank you for your modifications. 
Still, I am not sure of the place where to add the "Skip" part in the tutorial: it is currently at the end of the introduction (before the agenda) when I would have expected it to be just before the time-consuming steps (the XCMS steps in fact, the conversion from raw to mzML being sufficiently fast). Moreover, the convertion part and data format issue is an important point (not covered in the LC-MS XCMS tutorial), so keeping it as mandatory would be a good idea. In addition, in the "Import the data into Galaxy" there are some precious comments about data (e.g. the comment about the extra files), so it would be a pity to enable trainees to skip it just because of computing time. 

In conclusion, I would suggest to move the "Skip" part from introduction to the "Peak detection using XCMS" part (between the first paragraph and the "Peak picking" part), and to reduce its content to the "XCMS" object only, mentioning explicitly that this corresponds to the final dataset from the XCMS step (going from "Peak picking" to "Integrating areas of missing peaks") that can be used to save time if processing takes too long and/or if the trainee already know the XCMS steps and wants to proceed with the GC-specific parts. 

I hope this time I am at least a bit clearer than my previous suggestion but that is not easy to explain. So if needed, do not hesitate to contact me by e-mail and then we can plan a short call to discuss it.

Reply:
That's an excellent idea! I moved the "Skip" part to the suggested location in [7e2b8c4](https://github.com/galaxyproject/training-material/pull/3949/commits/7e2b8c4de1830b4537a90de1ed1636c86a9e2b24). I think you are right that this way, the trainee does not skip some important reading. Thank you for the detailed description.